app/email/request-email.js

```diff
@@ -0,0 +1,102 @@
+import { createFileName } from '../document/create-filename.js'
+import { appConfig } from '../config/index.js'
+import { DOCUMENT_STATUSES, NEW_USER } from '../constants.js'
+import { update } from '../repositories/document-log-repository.js'
+import appInsights from 'applicationinsights'
+import { sendSFDEmailMessage } from './sfd-client.js'
+
+const {
+  applyServiceUri, claimServiceUri, templateIdFarmerApplicationGenerationNewUser,
+  templateIdFarmerApplicationGenerationExistingUser
+} = appConfig
+
+const AddressType = {
+  ORG_EMAIL: 'orgEmail',
+  EMAIL: 'email',
+  CC: 'CC'
+}
+
+const sendEmailRequest = async (requestParams) => {
+  const { logger, emailAddress, personalisation, reference, crn, sbi, templateId, addressType } = requestParams
+
+  let success = false
+  try {
+    await sendSFDEmailMessage(logger,
+      templateId,
+      emailAddress,
+      personalisation,
+      reference,
+      crn,
+      sbi
+    )
+
+    logger.info(`Request sent to sfd message proxy for ${reference} (${addressType})'`)
+
+    appInsights.defaultClient.trackEvent({
+      name: 'agreement-email-requested',
+      properties: {
+        status: success,
+        reference,
+        addressType,
+        templateId
+      }
+    })
+    success = true
+  } catch (e) {
+    logger.error(`Error occurred requesting email send for ${reference}. Error: ${JSON.stringify(e.response?.data)}`)
```

Comment:
not better to log this as an object passed to the logger rather than stringifying?

Reply:
Was just a leftover relic from this file's former life really. I've simplified it - the only exception we can get coming out of there now anyway realistically is a simple Error object with the message 'SFD Validation Failed'